### PR TITLE
Kill infra pod when a container is restarted.

### LINKED ISSUE #4759: Kubelet kills POD container after other containers when the containers have changed
cAdvisor API is not accessible from the node. It is accessible from inside the cadvisor container though. This is the reason for the cadvisor e2e test to fail and the recent heapster errors reported by users.
The cadvisor version itself hasn't changed recently though.

### PR DESCRIPTION
**Pull request by vmarmol**
Revert 0e20f7d73659c8dcec771f8d72f525360c26fcc3.
Fixes #4759.

### CODE REVIEW

**Comment by vmarmol** [comment]
Don't merge yet. Running e2e to verify

/cc @dchen1107

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -1131,6 +1129,16 @@ func (kl *Kubelet) syncPod(pod *api.BoundPod, dockerContainers dockertools.Docke
 				continue
 			}
 			killedContainers[containerID] = empty{}
+
+			if podChanged {
+				// Also kill associated pod infra container if the pod changed.
+				if podInfraContainer, found, _ := dockerContainers.FindPodContainer(podFullName, uid, dockertools.PodInfraContainerName); found {
+					if err := kl.killContainer(podInfraContainer); err != nil {
+						glog.V(1).Infof("Failed to kill pod infra container %q: %v", podInfraContainer.ID, err)
```
Comment (vishh) [inline pkg/kubelet/kubelet.go]: Should this be a VLOG?  Should we generate events for such failures?

Comment (vmarmol) [inline pkg/kubelet/kubelet.go]: This was a log before and a temp fix, we're gonna work on a more permanent fix. Today we don't surface these failures as events just yet.

**Comment by vmarmol** [comment]
Manual tests show this fixes the issue. I'm having trouble getting the e2e cluster up for some reason...

**Comment by vmarmol** [comment]
cAdvisor e2e test is passing. I think we're good to go @dchen1107

**Comment by vishh** [comment]
LGTM.

**Comment by vmarmol** [comment]
All e2e tests are passing too

**Comment by dchen1107** [comment]
Thanks for the fix.

**Pull request merged**